bee-network/src/swarm/protocols/gossip/gossip.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::GossipHandler;
+
+use crate::network::meta::{ConnectionInfo, Origin};
+
+use libp2p::{
+    core::{connection::ConnectionId, ConnectedPoint},
+    swarm::{NegotiatedSubstream, NetworkBehaviour, NetworkBehaviourAction, ProtocolsHandler},
+    Multiaddr, PeerId,
+};
+use tokio::sync::mpsc;
+
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::atomic::{AtomicBool, Ordering},
+    task::Poll,
+};
+
+// false == inbound
+pub static GOSSIP_ORIGIN: AtomicBool = AtomicBool::new(false);
```

Comment:
What's this static for?

Reply:
It's the part that needs to be attributed with FIXME. The issue with it is minor enough to not deal with it in the current release, but in a future (maybe already next patch). The story why I need it atm is long and sad. I am sure I find a better way and can get rid of it. I'll add a FIXME.

I added further context to the problem in the code.